privilege/privileges/cache.go

```diff
@@ -357,12 +373,7 @@ func (p *MySQLPrivilege) LoadRoleGraph(ctx sessionctx.Context) error {
 
 // LoadUserTable loads the mysql.user table from database.
 func (p *MySQLPrivilege) LoadUserTable(ctx sessionctx.Context) error {
-	userPrivCols := make([]string, 0, len(mysql.Priv2UserCol))
```

Comment:
I would like to keep this dynamic array. Won't we add more privUserCol in the future?

Reply:
In theory actually it should become rare. The new privileges added to MySQL are usually added with something called `DYNAMIC` privileges. We have a spec for adding it here: https://github.com/pingcap/tidb/pull/22486

We have already extended the privileges table prior to add ad-hoc DYNAMIC privileges. I don't think we need to migrate these yet, but it does actually break MySQL compatibility since a mysqldump becomes non-restorable.

Reply:
It makes sense to me, though. I wonder that there should be a comment at the `mysql.Priv2UserCol` to remind the update of tidb.

It is also rare that internal table names will cause problems, even if they are using string concat. The new API could handle the problem, too.

Reply:
Lets see what @tiancaiamao thinks. I am happy to take either approach, but prefer the comment if he is okay with that.